### PR TITLE
EditorBeatmap's TimelineZoom should never be zero

### LINKED ISSUE #20306: Beatmap's `TimelineZoom` should never be zero
### Type

Cosmetic

### Bug description

As seen in https://github.com/ppy/osu/discussions/20238, having a `TimelineZoom` of 0 results in all hitobjects being stacked at one time in the editor.

Not only does this result in bad performance, it also crashes as soon as you do anything with the timeline - dragging blueprints, changing the zoom values, and possibly more.

### Screenshots or videos

_No response_

### Version

90e5949e9fdafc88bd3376c968b6dc5dd9a0426e

### Logs

-

### PR DESCRIPTION
**Pull request by ondrejvajdak**
fixes #20306

### CODE REVIEW

File: osu.Game/Screens/Edit/EditorBeatmap.cs [hunk 1]
```diff
@@ -96,6 +96,7 @@ public EditorBeatmap(IBeatmap playableBeatmap, ISkin beatmapSkin = null, Beatmap
             PlayableBeatmap.ControlPointInfo = ConvertControlPoints(PlayableBeatmap.ControlPointInfo);
 
             this.beatmapInfo = beatmapInfo ?? playableBeatmap.BeatmapInfo;
+            if (this.beatmapInfo.TimelineZoom <= 0) this.beatmapInfo.TimelineZoom = 1;
```
Comment (frenzibyte) [inline osu.Game/Screens/Edit/EditorBeatmap.cs:99]: Don't know about that. Are you aware that there are limits to zooming in the timeline? Surely zero isn't in that range, but it still ended up being used, which is not correct behaviour.

Check how zooming is set up in `Timeline`, that should be enough to tell you what's actually wrong.

**Review by frenzibyte — CHANGES_REQUESTED** [review]
As proposed.

**Comment by Drison64** [comment]
Hey, I updated it, but not sure if it's what you meant. Would you mind explaining what you had in mind if it isn't?

**Comment by frenzibyte** [comment]
That's precisely what I had in mind. In addition, would add a validity check in `SetupZoom` that the initial zoom is between minimum and maximum values, so that this doesn't happen again.

File: osu.Game/Screens/Edit/Compose/Components/Timeline/Timeline.cs [hunk 2]
```diff
@@ -196,7 +196,7 @@ protected override void Update()
             {
                 defaultTimelineZoom = getZoomLevelForVisibleMilliseconds(6000);
 
-                float initialZoom = (float)(defaultTimelineZoom * editorBeatmap.BeatmapInfo.TimelineZoom);
+                float initialZoom = (float)(defaultTimelineZoom * (editorBeatmap.BeatmapInfo.TimelineZoom <= 0 ? 1 : editorBeatmap.BeatmapInfo.TimelineZoom));
```
Comment (frenzibyte) [inline osu.Game/Screens/Edit/Compose/Components/Timeline/Timeline.cs:199]: There's no negative timeline zoom, that's guaranteed by `LegacyBeatmapDecoder`.
```suggestion
                float initialZoom = (float)(defaultTimelineZoom * (editorBeatmap.BeatmapInfo.TimelineZoom == 0 ? 1 : editorBeatmap.BeatmapInfo.TimelineZoom));
```

**Review by peppy — APPROVED** [review]
Seems okay, hopefully the new exception doesn't throw at runtime.

**Pull request merged**